sparse_strips/vello_cpu/tests/gradient.rs

```diff
@@ -0,0 +1,1229 @@
+// Copyright 2025 the Vello Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use vello_common::kurbo::{Affine, Point, Rect};
+use vello_common::peniko::GradientKind;
+use vello_cpu::paint::Gradient;
+use crate::util::{check_ref, get_ctx, stops_blue_green_red_yellow, stops_green_blue};
+
+pub(crate) fn tan_45() -> f64 {
+    45.0_f64.to_radians().tan()
+}
+
+#[test]
+fn gradient_on_3_wide_tiles() {
+    let mut ctx = get_ctx(600, 32, false);
+    let rect = Rect::new(4.0, 4.0, 596.0, 28.0);
+
+    let gradient = Gradient {
+        kind: GradientKind::Linear {
+            start: Point::new(0.0, 0.0),
+            end: Point::new(600.0, 0.0),
+        },
+        stops: stops_green_blue(),
+        transform: Affine::IDENTITY,
+        extend: vello_common::peniko::Extend::Pad,
+        opacity: 1.0,
+    };
+
+    ctx.set_paint(gradient);
+    ctx.fill_rect(&rect);
+
+    check_ref(&ctx, "gradient_on_3_wide_tiles");
+}
+
+#[test]
+fn gradient_with_global_alpha() {
+    let mut ctx = get_ctx(100, 100, false);
+    let rect = Rect::new(10.0, 10.0, 90.0, 90.0);
+
+    let gradient = Gradient {
+        kind: GradientKind::Linear {
+            start: Point::new(10.0, 0.0),
+            end: Point::new(90.0, 0.0),
+        },
+        stops: stops_blue_green_red_yellow(),
+        transform: Affine::IDENTITY,
+        extend: vello_common::peniko::Extend::Pad,
+        opacity: 0.5,
+    };
+
+    ctx.set_paint(gradient);
+    ctx.fill_rect(&rect);
+
+    check_ref(&ctx, "gradient_with_global_alpha");
+}
+
+mod linear {
```

Comment:
Let's add a test case for a purely vertical linear gradient, e.g. from (0, 0) to (0, 90)